game/src/js/scenes/TitleScene.js

```diff
@@ -41,15 +40,35 @@ export default class TitleScene extends BaseScene {
     });
 
     //controls, credits, offline mode buttons
-    let settingsButton = new Button(this, 450, 350, 'SETTINGS');
-    settingsButton.buttonDown(() => {
-      new DOMModal(this, 'settings', {
-        cancelButtonSelector: '.exit',
-        onCancel: (modal) => {
-          modal.close();
-        }
+    if(ServiceWorker.isSupported()) {
+      const assets = this.cache.json.get('assets');
+      let serviceWorker = new ServiceWorker();
+      let settingsButton = new Button(this, 450, 350, 'SETTINGS', serviceWorker.isRegistered());
+      let offlineMode = false;
+      settingsButton.buttonDown(() => {
+        new DOMModal(this, 'settings', {
+          cancelButtonSelector: '.exit',
+          acceptButtonSelector: '#settingsCheck',
+          onCancel: (modal) => {
+
+            modal.close();
+          },
+          onAccept: (modal) => {
+              offlineMode = true;
+            
+              if (modal.modal.querySelector('#settingsCheck').checked) {
```

Comment:
Indentation is a bit weird here, so it took me a bit to understand where it all started and ended.